pkg/apis/storage/types.go

```diff
@@ -424,3 +445,77 @@ type CSINodeList struct {
 	// items is the list of CSINode
 	Items []CSINode
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// CSIStorageCapacity stores the result of one CSI GetCapacity call.
+// For a given StorageClass, this describes the available capacity in a
+// particular topology segment.  This can be used when considering where to
+// instantiate new PersistentVolumes.
+//
+// For example this can express things like:
+// - StorageClass "standard" has "1234 GiB" available in "topology.kubernetes.io/zone=us-east1"
+// - StorageClass "localssd" has "10 GiB" available in "kubernetes.io/hostname=knode-abc123"
+//
+// The following three cases all imply that no capacity is available for
+// a certain combination:
+// - no object exists with suitable topology and storage class name
+// - such an object exists, but the capacity is unset
```

Comment:
Is unset valuable enough to keep as yet another mode that consumers must be aware of?  Why not just default that to 0?

Reply:
Right now it isn't particularly important. This will become more relevant when adding other kinds of capacity ("total capacity", "maximum volume size") because then leaving "capacity" at nil seems suitable.